runtime/src/sorted_storages.rs

```diff
@@ -0,0 +1,117 @@
+use crate::accounts_db::SnapshotStorage;
+use log::*;
+use solana_measure::measure::Measure;
+use solana_sdk::clock::Slot;
+use std::ops::Range;
+
+pub struct SortedStorages<'a> {
+    range: Range<Slot>,
+    storages: Vec<Option<&'a SnapshotStorage>>,
+    count: usize,
+}
+
+impl<'a> SortedStorages<'a> {
+    pub fn get(&self, slot: Slot) -> Option<&SnapshotStorage> {
+        if !self.range.contains(&slot) {
+            None
+        } else {
+            let index = (slot - self.range.start) as usize;
+            self.storages[index]
+        }
+    }
+
+    pub fn range_width(&self) -> u64 {
+        self.range.end - self.range.start
+    }
+
+    pub fn range(&self) -> &Range<Slot> {
+        &self.range
+    }
+
+    pub fn len(&self) -> usize {
+        self.count
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    pub fn new(source: &'a [SnapshotStorage]) -> Self {
+        let mut min = Slot::MAX;
+        let mut max = Slot::MIN;
+        let mut count = 0;
+        let mut time = Measure::start("get slot");
+        let slots = source
+            .iter()
+            .map(|storages| {
+                count += storages.len();
+                if !storages.is_empty() {
+                    storages.first().map(|storage| {
```

Comment:
Is it correct that this code assumes there's only one `AccountsStorageEntry` per `storages` here? I think that's fine with the AccountsCache, yes, but does this code (need to) get used where AccountsCache is disabled? Maybe this assumption can be commented with `///` on the function and an assert?

Reply:
No. This code assumes that everything in 'storages' (line 46) has the same answer for .slot() on line 50. first() is like storages[0] but returns None if storages is empty.

Reply:
Oh duh, this is only getting the slots, not what's in each of these storages. Thanks!